#283: Condensed folder picker

mdrlzy opened the pull request:
#283

kirillt commented:
Update erased folders, I guess that is expected since we changed path of roots list storage, but:
- [ ] Auto-discovery was not suggested after the update.
- [ ] Manual addition of a root causes RoomDB exception:
http://65.21.4.137:8080/app/1/bug/95/report/93eb90c9-f755-493f-a684-12165d2cc6e2

After manual data erasure in settings menu:
- [ ] Auto-discovery triggers indexing of each root, which makes user wait too long.

mdrlzy commented:
> Auto-discovery was not suggested after the update.

Added rescan roots button to the settings

> Manual addition of a root causes RoomDB exception

I forgot to change database version, also added small manual migration to avoid destructive migration and unnecessary indexing

> After manual data erasure in settings menu:
> Auto-discovery triggers indexing of each root, which makes user wait too long.

You also deleted the app database, so you need to reindex roots

kirillt commented:
> You also deleted the app database, so you need to reindex roots

Yes, but we can do it lazily, waiting till particular root will be opened, isn't it? This way, user would only wait every time she opens a root for first time, but would not wait at all during opening the app.

> Added rescan roots button to the settings

:+1: 

> I forgot to change database version, also added small manual migration to avoid destructive migration and unnecessary indexing

Will test update again.
